src/d3a/models/strategy/home_meter.py

```diff
@@ -258,14 +271,13 @@ def area_reconfigure_event(self, *args, **kwargs):
         """
         self._area_reconfigure_consumption_prices(**kwargs)
         self._area_reconfigure_production_prices(**kwargs)
+        self.offer_update.update_and_populate_price_settings(self.area)
+        self.bid_update.update_and_populate_price_settings(self.area)
 
         # Update the raw profile. It will be read later while setting the energy forecast.
         if kwargs.get("home_meter_profile") is not None:
             self.home_meter_profile = kwargs["home_meter_profile"]
-
-        self.offer_update.update_and_populate_price_settings(self.area)
-        self.bid_update.update_and_populate_price_settings(self.area)
-        self._set_energy_forecast_for_future_markets(reconfigure=True)
+            self._set_energy_forecast_for_future_markets(reconfigure=True)
```

Comment:
Just a doubt: what happens if the `home_meter_profile` is `None` here? Can this happen?

Reply:
yes, indeed, this can happen. But is caught by the if clause. We only have to run `self._set_energy_forecast_for_future_markets` when it is not None.

You gave me a hint for something that is missing still:
What if the user wants to change the profile (via the profile_uuid). 
I need to adapt for this case (not only here but also in the predefined load and pv. Thanks!

Reply:
Did that here: https://github.com/gridsingularity/d3a/pull/1152/commits/845adce0cda917b1d0e1712cd6e39c4256d17e4d